### ISSUE TITLE
Sveltos gives for granted that Control Plane is made of machines

### ISSUE THREAD

**Issue by prometherion** [projectsveltos/libsveltos#195]
We tested Sveltos with [Kamaji[(https://github.com/clastix/kamaji) which is running Control Plane as pods instead of virtual machines.

With that said, a Cluster API-based cluster will never be considered ready by Sveltos for this:
https://github.com/projectsveltos/libsveltos/blob/2074a30e9c999462fab494153016a45f244c9991/lib/clusterproxy/clusterproxy.go#L210-L233

I would suggest, rather, adhering to the CAPI status contracts, such as `cluster.status.controlPlaneReady` field.

```
KIND:     Cluster
VERSION:  cluster.x-k8s.io/v1beta1

FIELD:    controlPlaneReady <boolean>

DESCRIPTION:
     ControlPlaneReady defines if the control plane is ready.
```

**Comment by prometherion** [comment]
My bad, duplicate of #193.

@gianlucam76 please, feel free to close, as well as my proposed fix.

**Comment by gianlucam76** [comment]
Thanks @prometherion 

I missed you already had created this one. Description here is more clear than the issue I had created. 

I will keep this open and will mark resolved only once I get a chance to verify with Kamaji so to be sure there is nothing else left.

Thank you for opening this one.

**Comment by gianlucam76** [comment]
This is verified and working now. So closing this issue. Thank you.
